### PR TITLE
build: add Apache Cloudberry build automation and test framework

### PR DESCRIPTION
**Pull request by edespino**
Introduces a comprehensive build automation framework for Apache Cloudberry:

- Configure/build/test automation scripts with standardized logging
- Unit test and regression test execution framework
- Demo cluster creation and management utilities
- Test results parsing and reporting tools
- Shared utility functions for common operations

The framework provides:
- Consistent environment setup and teardown
- Standardized logging and error handling
- Clear exit codes and error reporting
- GitHub Actions integration support
- RPM package building capabilities

This commit establishes the foundation for Apache
Cloudberry (incubating)
CI/CD pipeline and release automation.

### CODE REVIEW

File: build_automation/cloudberry/scripts/build-cloudberry.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.  See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# --------------------------------------------------------------------
+#
+# Script: build-cloudberry.sh
+# Description: Builds CloudBerry DB from source code and installs it.
```
Comment (tuhaihe) [inline build_automation/cloudberry/scripts/build-cloudberry.sh:23]: `CloudBerry DB` ~> `Cloudberry` ?

File: build_automation/cloudberry/scripts/build-cloudberry.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+#             Performs the following steps:
+#             1. Builds main CloudBerry DB components
```
Comment (tuhaihe) [inline build_automation/cloudberry/scripts/build-cloudberry.sh:25]: `CloudBerry` ~> `Cloudberry` ?

File: build_automation/cloudberry/scripts/build-cloudberry.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+# Source common utilities
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "${SCRIPT_DIR}/cloudberry-utils.sh"
+
+# Define log directory and files
+export LOG_DIR="${SRC_DIR}/build-logs"
+BUILD_LOG="${LOG_DIR}/build.log"
+
+# Initialize environment
+init_environment "CloudBerry Build Script" "${BUILD_LOG}"
```
Comment (tuhaihe) [inline build_automation/cloudberry/scripts/build-cloudberry.sh:68]: `CloudBerry` ~> `Cloudberry`

File: build_automation/cloudberry/scripts/configure-cloudberry.sh [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+# Source common utilities
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "${SCRIPT_DIR}/cloudberry-utils.sh"
+
+# Define log directory and files
+export LOG_DIR="${SRC_DIR}/build-logs"
+CONFIGURE_LOG="${LOG_DIR}/configure.log"
+
+# Initialize environment
+init_environment "CloudBerry Configure Script" "${CONFIGURE_LOG}"
```
Comment (tuhaihe) [inline build_automation/cloudberry/scripts/configure-cloudberry.sh:90]: `CloudBerry` ~> `Cloudberry`

File: build_automation/cloudberry/scripts/build-cloudberry.sh (hunk 1 above)
Comment (edespino) [reply build_automation/cloudberry/scripts/build-cloudberry.sh:23]: @tuhaihe  I will change the series of them scattered throughout.

File: build_automation/cloudberry/scripts/configure-cloudberry.sh [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+# Initial setup
+log_section "Initial Setup"
+execute_cmd sudo rm -rf /usr/local/cloudberry-db || exit 2
+execute_cmd sudo chmod a+w /usr/local || exit 2
+execute_cmd mkdir -p /usr/local/cloudberry-db/lib || exit 2
+execute_cmd sudo cp /usr/local/xerces-c/lib/libxerces-c.so \
+        /usr/local/xerces-c/lib/libxerces-c-3.3.so \
+        /usr/local/cloudberry-db/lib || exit 3
+execute_cmd sudo chown -R gpadmin.gpadmin /usr/local/cloudberry-db || exit 2
```
Comment (tuhaihe) [inline build_automation/cloudberry/scripts/configure-cloudberry.sh:100]: `gpadmin.gpadmin` ~> `gpadmin:gpadmin`? Not sure about this, FYI.

**Review by tuhaihe — APPROVED** [review]
Just left some non-tech comments, FYI. Thanks!

**Review by tuhaihe — APPROVED** [review]

**Pull request merged**